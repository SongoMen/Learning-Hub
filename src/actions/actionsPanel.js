export const SET_POPUP = "SET_POPUP";
export const CHANGE_RIGHT_BAR = "CHANGE_RIGHT_BAR";


export function SetPopupAvatar(status) {
  return {
    type: SET_POPUP,
    payload: status
  };
}

export function ChangeRightBar(status) {
  return {
    type: CHANGE_RIGHT_BAR,
    payload: status
  };
}