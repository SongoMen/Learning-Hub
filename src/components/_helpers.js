import React from "react";
import { Link } from "react-router-dom";

export function Logo(props) {
  console.log(typeof props.link)
  return (
    <Link to={typeof props.link === "undefined" ? "/" : "#"}>
      <div className={props.class}>
        <svg
          xmlns="http://www.w3.org/2000/svg"
          width="24"
          height="24"
          fill="none"
          stroke="currentColor"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="2"
          className="logo"
          viewBox="0 0 24 24"
        >
          <path d="M16.5 9.4L7.5 4.21"></path>
          <path d="M21 16V8a2 2 0 00-1-1.73l-7-4a2 2 0 00-2 0l-7 4A2 2 0 003 8v8a2 2 0 001 1.73l7 4a2 2 0 002 0l7-4A2 2 0 0021 16z"></path>
          <path d="M3.27 6.96L12 12.01 20.73 6.96"></path>
          <path d="M12 22.08L12 12"></path>
        </svg>
        <h3>Learning Hub</h3>
      </div>
    </Link>
  );
}
