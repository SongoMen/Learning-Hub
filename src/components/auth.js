import * as firebase from "firebase/app";
import "firebase/auth";
import "firebase/firestore";
import "firebase/performance";
import 'firebase/analytics';

import config from "./firebaseAuth"

firebase.initializeApp(config);
firebase.analytics();

export const googleProvider = new firebase.auth.GoogleAuthProvider();
export const firebaseAuth = firebase.auth;
export const db = firebase.firestore();

export function auth(email, pw) {
  let username = localStorage.getItem("user");
  return firebaseAuth()
    .createUserWithEmailAndPassword(email, pw)
    .then(function(newUser) {
      db.collection("users")
        .doc(newUser.user.uid)
        .set({
          email,
          username,
          admin: false,
        })
        .catch(function(error) {
          console.error("Error writing document: ", error);
        });
      return firebase.auth().currentUser.updateProfile({
        displayName: username
      });
    });
}

export function logout() {
  return firebaseAuth().signOut();
}

export function login(email, pw) {
  return firebaseAuth().signInWithEmailAndPassword(email, pw);
}

export function resetPassword(email) {
  return firebaseAuth().sendPasswordResetEmail(email);
}
