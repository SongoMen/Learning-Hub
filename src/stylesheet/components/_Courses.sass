.Courses
    padding: 20px 30px
    width: 68%
    background-color: $panelColor
    transition: width .3s
    @media only screen and (max-width: 1270px) 
        width: 74%
    &.active
        width: 88%
        @media only screen and (max-width: 1270px) 
            width: 94%
    &__list 
        a
            float: left
            margin: 5px 10px
        svg
            width: auto!important