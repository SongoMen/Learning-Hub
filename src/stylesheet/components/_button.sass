.button
    stroke: $primaryColor
    background-color: $primaryColorbg
    padding: 8px
    border-radius: 5px
    transition: all .3s
    cursor: pointer
    &:hover
        background-color: $primaryColor
        stroke: white
    &.backwards
        transform: rotate(180deg)