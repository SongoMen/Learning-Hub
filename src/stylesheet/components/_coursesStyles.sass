.courses__box
    display: flex
    width: 300px
    border-radius: 10px
    justify-content: flex-start
    align-items: center
    height: 15%
    position: relative
    padding: 20px
    cursor: pointer
    margin: 25px 0
    min-width: 250px
    max-height: 250px
    white-space: nowrap
    svg:nth-child(1)
        width: 20%
        min-width: 40px
        border-radius: 10px
        padding: 10px
        margin-right: 20px
        fill: white
    .courses__arrow
        position: absolute
        bottom: 10px
        right: 15px
        width: 15px
        height: 15px
        transition: all .3s
    svg
        stroke: white
    &:hover
        .courses__arrow
            right: 5px
.english1
    background-color: #4e4cac
    svg:nth-child(1)
        width: 20%
        min-width: 40px
        background-color: #5b5ab3
        border-radius: 10px
        padding: 10px
        margin-right: 20px
        fill: white
    h5:nth-child(1)
        color: #babbff
        margin: 0

.jsfun
    background-color: #ff808b
    svg:nth-child(1)
        width: 20%
        min-width: 40px
        background-color: #ff919b
        border-radius: 10px
        padding: 10px
        margin-right: 20px
        fill: white
    h5:nth-child(1)
        color: #ffdadd
        margin: 0
    .courses__arrow
        position: absolute
        bottom: 10px
        right: 15px
        width: 15px
        height: 15px
        transition: all .3s
.courses__info
    display: flex
    justify-content: center
    right: 30px
    align-items: flex-start
    flex-direction: column
    h4
        margin: 0
        color: white
.courseLogo svg
    width: 20%
    min-width: 40px
    background-color: $accentColor2
    border-radius: 10px
    padding: 10px
    margin-right: 20px
    fill: white