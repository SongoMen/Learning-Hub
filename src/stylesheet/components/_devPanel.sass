.DevPanel
    padding: 20px 30px
    width: 68%
    background-color: $panelColor
    transition: width .3s
    &__list
        height: 79%
    &__topNav
        height: 10%
    &__title
        height: 10%
        margin: 0
    .form-btn
        min-height: 40px
        width: 20%
    &__edit
        height: 100%
        width: 100%
        .arrow
            float: right
            width: 30px
            height: 30px
    &__lessons
        width: 100%
        height: 100%
        overflow: auto
        display: flex
        align-items: center
        flex-direction: column
    &__lesson
        width: 100%
        padding: 5px 20px
        background-color: $accentColor1
        display: flex
        justify-content: space-between
        align-items: center
        border-radius: 10px
        transition: all .3s
        cursor: pointer
        margin: 20px 0 
        svg
            margin-right: 10px
            transition: all .3s
        &:hover
            box-shadow: 0px 0px 0px 10px $accentColor2
            svg
                margin-right: 0
    .form-btn
        text-transform: uppercase
        h4
            margin: 0
.popupDev
    &__content
        width: 100%
        padding-top: 0px
        display: flex
        justify-content: center
        align-items: center
        flex-direction: column
        .form-btn
            margin: 20px
        .form-line
            width: 40%