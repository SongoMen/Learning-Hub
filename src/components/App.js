import React, { Component } from "react";
import {
  Route,
  BrowserRouter as Router,
  Redirect,
  Switch
} from "react-router-dom";
import { firebaseAuth } from "./auth";

import Loader from "./elements/Loader";
import LandingPage from "./LandingPage/LandingPage";
import Login from "./Login/Login";
import Register from "./Register/Register";
import Dashboard from "./Dashboard/Dashboard";
import CourseContainer from "./CoursePage/CourseContainer";

function PrivateRoute({ component: Component, authed, ...rest }) {
  return (
    <Route
      {...rest}
      render={props =>
        authed === true ? (
          <Component {...props} {...rest} />
        ) : (
          <Redirect
            to={{ pathname: "/login", state: { from: props.location } }}
          />
        )
      }
    />
  );
}

function PublicRoute({ component: Component, authed, ...rest }) {
  return (
    <Route
      {...rest}
      render={props =>
        authed === false ? (
          <Component {...props} />
        ) : (
          <Redirect to="/dashboard" />
        )
      }
    />
  );
}

class App extends Component {
  _isMounted = false;
  state = {
    authed: false,
    loading: true,
    theme: ""
  };
  componentDidMount() {
    this._isMounted = true;
    this.removeListener = firebaseAuth().onAuthStateChanged(user => {
      if (this._isMounted) {
        if (user) {
          this.setState({
            authed: true,
            loading: false
          });
        } else {
          this.setState({
            authed: false,
            loading: false
          });
        }
      }
    });
  }
  componentWillUnmount() {
    this._isMounted = false;
    this.removeListener();
  }

  render() {
    return this.state.loading ? (
      <Loader />
    ) : (
      <Router>
        <div className="container">
          <Switch>
            <Route exact path="/" component={LandingPage} />
            <PublicRoute
              authed={this.state.authed}
              path="/login"
              component={Login}
            />
            <PublicRoute
              authed={this.state.authed}
              path="/register"
              component={Register}
            />
            <PrivateRoute
              authed={this.state.authed}
              path="/dashboard"
              component={Dashboard}
            />
            <PrivateRoute
              authed={this.state.authed}
              path="/courses"
              component={Dashboard}
            />
            <PrivateRoute
              authed={this.state.authed}
              path="/devpanel"
              component={Dashboard}
            />
            <PrivateRoute
              authed={this.state.authed}
              path="/course/:name"
              component={CourseContainer}
              exact
              name={window.location.pathname.split("/")[2]}
            />
            <PrivateRoute
              authed={this.state.authed}
              path="/course/:name/:id"
              component={CourseContainer}
              id={window.location.pathname.split("/")[3]}
            />
          </Switch>
        </div>
      </Router>
    );
  }
}

export default App;
