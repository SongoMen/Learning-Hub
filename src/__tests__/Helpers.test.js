import * as helpers from "../components/_helpers";
import * as actions from "../actions/actionsPanel";
import renderer from "react-test-renderer";
import React from "react";
import { BrowserRouter as Router } from "react-router-dom";

describe("Test helpers", () => {
  test("input renders correctly", () => {
    const component = renderer.create(<helpers.Input/>);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
