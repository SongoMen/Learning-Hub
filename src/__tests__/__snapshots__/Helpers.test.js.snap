// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test helpers input renders correctly 1`] = `
<input
  className="input"
  onChange={[Function]}
  required={true}
/>
`;

exports[`Test helpers logo renders correctly 1`] = `
<input
  className="input"
  onChange={[Function]}
  required={true}
/>
`;

exports[`Test helpers mask renders correctly 1`] = `
<div>
  <div
    className="mask"
  />
  <nav
    className="Nav-bar"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <div
        className="Nav-bar__logo"
      >
        <svg>
          logo.svg
        </svg>
        <h3>
          Learning Hub
        </h3>
      </div>
    </a>
  </nav>
</div>
`;
